src_c/geometry.c

```diff
@@ -8,6 +8,106 @@
 
 #define PYGAMEAPI_GEOMETRY_NUMSLOTS 21
 
+static PyObject *
+pg_raycast(PyObject *_null, PyObject *const *args, Py_ssize_t nargs)
+{
+    PyObject **farr;
+    Py_ssize_t farr_length;
+    Py_ssize_t loop;
+    double max_t;
+    pgLineBase line;
+
+    if (nargs == 4) {
+        if (!pg_TwoDoublesFromObj(args[0], &line.x1, &line.y1)) {
+            return RAISE(PyExc_TypeError,
+                         "the origin requires a pair of floats");
+        }
+        double max_dist;
+        if (!pg_DoubleFromObj(args[2], &max_dist)) {
+            return RAISE(PyExc_TypeError, "invalid max distance");
+        }
+        if (max_dist < 0) {
+            max_dist = DBL_MAX;
+        }
+        if (!pg_TwoDoublesFromObj(args[1], &line.x2, &line.y2)) {
+            double angle;
+            if (!pg_DoubleFromObj(args[1], &angle)) {
+                return RAISE(PyExc_TypeError,
+                             "expected a pair of floats or a single float");
```

Comment:
You don't expect a pair of floats from an angle parameter... Rephrase this error message to something like: `"Invalid ray angle value, must be numeric"`